eth/api.go

```diff
@@ -317,6 +319,24 @@ type BlockTraceResult struct {
 	Error      string                `json:"error"`
 }
 
+// TraceArgs holds extra parameters to trace functions - either a vm.LogConfig, or a
+// string to evaluate as a JavaScript object.
+type TraceArgs struct {
+	logConfig  *vm.LogConfig
+	tracer     string
+}
+
+func (ta *TraceArgs) UnmarshalJSON(data []byte) error {
+	input := strings.TrimSpace(string(data))
+
+	if len(input) > 2 && input[0] == '"' && input[len(input) - 1] == '"' {
+		return json.Unmarshal(data, &ta.tracer)
+	} else {
+		ta.logConfig = &vm.LogConfig{}
+		return json.Unmarshal(data, ta.logConfig)
+	}
+}
+
 // TraceBlock processes the given block's RLP but does not import the block in to
 // the chain.
 func (api *PrivateDebugAPI) TraceBlock(blockRlp []byte, config *vm.LogConfig) BlockTraceResult {
```

Comment:
Yes, but since they have different structure for their return values, I wanted to write that up as a separate (hopefully simple) PR.